components/CustomModal/modal.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react"
+import Image from "react-bootstrap/Image"
+
+import { ACTION_TYPE, ACTIONS } from "../../pages/merchant/_reducer"
+import styles from "./modal.module.css"
+
+interface Props {
+  children?: React.ReactNode
+  ref?: React.LegacyRef<HTMLDivElement>
+  isOpened?: boolean
+  modalTitle?: string
+  modalWidth?: string
+  dispatch: React.Dispatch<ACTION_TYPE>
+}
+
+const Modal = React.forwardRef(
+  (
+    { children, isOpened, modalTitle, modalWidth, dispatch }: Props,
+    ref: React.ForwardedRef<HTMLDivElement>,
+  ) => {
+    const defaultModalWidth = modalWidth || "100%"
+    const defaultRef = React.useRef(null)
+    const modalFocus = ref || defaultRef.current
+
+    const [isOpen, setIsOpen] = React.useState<boolean>(isOpened || true)
+
+    const handleClose = () => {
```

Comment:
The name CustomModal implies that this component can be reusued as a generic modal, however ```handleClose``` forces this modal to be specific to the use case of being a PinnedToHomescreenModal. Either pass the ```handleClose``` in as a prop, or make it clear this is not a generic component.